### ISSUE TITLE
Change system and pointer APIs to pervasively use the `Range` type

### ISSUE THREAD

**Issue by k-sareen** [titzer/virgil#132]
Currently APIs such as `Ptr.atContents` and `System.file{Read,Write}K` assume the input will be an `Array<byte>`. Changing to a `Range<byte>` would make it more general.

**Comment by titzer** [comment]
I have done some work on this recently, e.g. `Pointer.atContents` can work with ranges now.

For the built-in `System` component, I was thinking of adding new methods that take a `Range<byte>` and then deprecating the old ones. This requires a stable rev, as they must first be introduced into `v3i` and then (with the `Pointer.atContents` change above integrated into stable) added the native implementations after the rev.

**Comment by k-sareen** [comment]
Yes that makes sense. Sorry for being AWOL again -- just have too many deadlines. Will be more free after mid-April. Might try my hand at making an LSP server again.
